plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -35,15 +34,19 @@ struct session_base {
 
 class send_update_send_queue_entry : public send_queue_entry_base {
    std::shared_ptr<session_base> session;
-   const chain::block_state_legacy_ptr block_state;
+   const chain::signed_block_ptr block;
+   const chain::block_id_type id;
+   uint32_t block_num;
 public:
-   send_update_send_queue_entry(std::shared_ptr<session_base> s, chain::block_state_legacy_ptr block_state)
+   send_update_send_queue_entry(std::shared_ptr<session_base> s, chain::signed_block_ptr block, chain::block_id_type id, uint32_t block_num)
          : session(std::move(s))
-         , block_state(std::move(block_state)){}
+         , block(std::move(block))
+         , id(std::move(id))
```

Comment:
id doesn't move, should pass by reference.

Reply:
Thanks. Fixed.